src/DotNetWorker.Core/Pipeline/FunctionExecutionMiddleware.cs

```diff
@@ -1,22 +1,24 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
+using System;
 using System.Threading.Tasks;
 using Microsoft.Azure.Functions.Worker.Invocation;
 
 namespace Microsoft.Azure.Functions.Worker.Pipeline
 {
-    internal class FunctionExecutionMiddleware
+    internal class FunctionExecutionMiddleware(IFunctionExecutor functionExecutor)
     {
-        private readonly IFunctionExecutor _functionExecutor;
-
-        public FunctionExecutionMiddleware(IFunctionExecutor functionExecutor)
-        {
-            _functionExecutor = functionExecutor;
-        }
+        private readonly IFunctionExecutor _functionExecutor = functionExecutor
+            ?? throw new ArgumentNullException(nameof(functionExecutor));
 
         public Task Invoke(FunctionContext context)
         {
+            if (context.Features.Get<IFunctionExecutor>() is { } executor)
```

Comment:
So, extensions authors (or even function app developers) have to set their `IFunctionExecutor` implementation in a middleware which gets executed before `FunctionExecutionMiddleware` per invocation? Or is there another way these can be registered (like during app bootstrapping? )

Reply:
Yes, it is set during middleware. FuctionExecutionMiddleware should always be the _last_ middleware in the pipeline, so any customer middleware will always have a chance to set this.

We will not support some other way to register this during app bootstraping - middleware will be the only way. Which I am happy with - this is the most flexible for customers and honestly not much work for them.